crates/profile/models/security-structures/src/roles_matrices_structures/security_shield_builder_rule_violation_reason.rs

```diff
@@ -3,15 +3,15 @@ use crate::prelude::*;
 pub trait AsShieldBuilderViolation {
     fn as_shield_validation(
         &self,
-    ) -> Option<SecurityShieldBuilderInvalidReason>;
+    ) -> Option<SecurityShieldBuilderRuleViolationReason>;
```

Comment:
RuleViolationReason is a bit of a tautology I think, I would drop the `Reason` suffix